src/catalog/src/system_schema/information_schema/region_peers.rs

```diff
@@ -176,9 +177,9 @@ impl InformationSchemaRegionPeersBuilder {
 
         let predicates = Predicates::from_scan_request(&request);
 
-        for schema_name in catalog_manager.schema_names(&catalog_name).await? {
+        for schema_name in catalog_manager.schema_names(&catalog_name, Mysql).await? {
```

Comment:
I think we should try to avoid hard code `Mysql` here. That's also why I suggest using `Option<QueryContextRef>` for this.

Reply:
I've tried use Option< QueryContext> though. We also need to query the CatalogManager in `pg_catalog` module. Same, we don't have QueryContext, so we have to pass `None`. There ambiguity comes. We don't know what to do when we get none(do pg resolving or mysql resolving)?

Reply:
My idea is we don't need to add default `query_context` when it's not from a query path. I'm afraid it will introduce more confusion with future maintenance. When the query context is `None`, we just fallback to default behaviour.

At the moment, there are only default behaviour and postgres behaviour:

- default: return information_schema
- postgres: return pg_catalog and information_schema, when table name equals `pg_class` (and maybe more like `pg_namespace`), return `<catalog>.pg_catalog.pg_class` table

Reply:
The point is we also don't have `query_context` whiling query catalog_manager inside pg_catalog. It's not correct to use the default behaviour under this scenario.

Reply:
oh, I got your point. When we are calling catalog_manager from pg_catalog, is there a chance we have `QueryContext` in the context that we can pass all the way to catalog_manager methods. If I understand correctly, the call must be initiated from a postgres query.

Reply:
Sorry for replying late. Unfortunately, it may hard to have `QueryContext` when calling catalog_manager methods. We only have `ScanRequest` when constructing a SystemTable to data stream. Check:
https://github.com/GreptimeTeam/greptimedb/blob/8eda36bfe3661c6355ef7ef71c2e6529e2166b53/src/catalog/src/system_schema.rs#L106